BUILD

```diff
@@ -1,3 +1,53 @@
-# This is a blank file. In Bazel, it is a convention and requirement to have a
-# blank file if you want to reference some path.
-# The reason this exists, is because this directory is referenced by workspaces
+load("@build_bazel_rules_apple//apple:swift.bzl", "swift_library")
+load("@build_bazel_rules_apple//apple:macos.bzl", "macos_command_line_application")
+
+objc_library(
+    name = "ObjcSupport",
+    srcs = glob(["Sources/ObjcSupport/*.m"]),
+    hdrs = glob(["Sources/ObjcSupport/include/*"]),
+    includes = ["Sources/ObjcSupport/include"]
+)
+
+# PodToBUILD is a core library enabling Skylark code generation
+swift_library(
+    name = "PodToBUILD",
+    srcs = glob(["Sources/PodToBUILD/*.swift"]),
+    deps = [":ObjcSupport"],
+    copts = ["-swift-version", "4", "-static-stdlib"],
+)
+
+# Compiler
+macos_command_line_application(
+    name = "Compiler",
+    minimum_os_version = "10.13",
+    deps = [":CompilerLib"],
+)
+
+swift_library(
+    name = "CompilerLib",
+    srcs = glob(["Sources/Compiler/*.swift"]),
+    deps = [":PodToBUILD"],
+)
+
+# RepoTools
+macos_command_line_application(
+    name = "RepoTools",
+    minimum_os_version = "10.13",
+    deps = [":RepoToolsLib"],
+)
+
+swift_library(
+    name = "RepoToolsLib",
+    srcs = glob(["Sources/RepoTools/*.swift"]),
+    deps = [":RepoToolsCore"],
+)
+
+swift_library(
+    name = "RepoToolsCore",
+    srcs = glob(["Sources/RepoToolsCore/*.swift"]),
+    deps = [":PodToBUILD"],
+    copts = ["-swift-version", "4", "-static-stdlib"],
+)
+
+# TODO: (jerry) Setup test suite
```

Comment:
we might as well do this in this PR if it isn't too much work

Reply:
This needs:
- test running setup
- trans SwiftCheck things https://github.com/typelift/SwiftCheck/blob/master/Package.swift

I don't see much value in setting this up ATM, other than consistency